pyne/utils.py

```diff
@@ -190,3 +192,166 @@ def remove(path):
         remove_tree(path, verbose=False)
     else:
         pass
+
+
+def str_to_unicode(s):
+    """
+    This function convert a str from binary or unicode to str (unicode).
+    If it is a list of string, convert every element of the list.
+
+    Parameters:
+    -----------
+    s : str or list of str
+
+    Returns:
+    --------
+    s : text str or list of unicode str
+    """
+    if isinstance(s, str) or isinstance(s, bytes):
+        # it is a str, convert to text str
+        try:
+            s = s.decode('utf-8')
+        except:
+            pass
+        return s
+    else:
+        for i, item in enumerate(s):
+            try:
+                s[i] = item.decode('utf-8')
+            except:
+                pass
+        return s    
+
+
+def is_close(a, b, rel_tol=1e-9, abs_tol=0.0):
+    return abs(a-b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol)
+
+
+def is_float(s):
+    """
+    This function checks whether a string can be converted as a float number.
+    """
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
+
+
+def str_almost_same(s1, s2, rel_tol=1e-9):
+    """
+    This function is used to compare two string to check whether they are
+    almost the same.
+    Return True if two strings are exactly the same.
+    Return True if two strings are almost the same with only slight difference
+    of float decimals.
+    Return False if two strings are different.
+    """
+    # if string can be converted to float number
+    if is_float(s1) and is_float(s2):
+        return is_close(float(s1), float(s2), rel_tol)
+    else:
+        # not a number
+        return s1 == s2
```

Comment:
could it be faster with a `try/catch` ?

Reply:
I am not quite understanding how to write the 'try/catch' for this function.
Do you mean use 'try/catch' to convert 's1' and 's2' to float? Like this?
```
try:
    s1, s2 = float(s1), float(s2)
    return return is_close(s1, s2, rel_tol)
catch:
    return s1 == s2
```